### PR TITLE
feat: Use native.js file to make react native really optional

### PR DESCRIPTION
**Pull request by doubleface**
Tested with the react native application
The native.js file will be ignored in other environment and shall cause no regression

This PR is made to replace https://github.com/cozy/cozy-client/pull/956 which did not work for connectors environment.

### CODE REVIEW

File: packages/cozy-client/src/authentication/mobile.native.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      }
+    }
+
+    Linking.addEventListener('url', linkListener)
+  })
+}
+
+/**
+ * Return the method to use for
+ * authentication based on the
```
Comment (JF-Cozy) [inline packages/cozy-client/src/authentication/mobile.native.js:60]: `authentication based on the` ... suspens ? :)

Comment (JF-Cozy) [reply packages/cozy-client/src/authentication/mobile.native.js:60]: Est-ce qu'il y a déjà des tests associés ?

Comment (Crash--) [reply packages/cozy-client/src/authentication/mobile.native.js:60]: C'est du hot fix et difficilement testable puisque du build grosso-modo

**Review by taratatach — APPROVED** [review]
Works on Cozy Desktop :+1:

**Pull request merged**